internal/indexer/data/mongo_client.py

```
@@ -0,0 +1,205 @@
+
+            logger.info("Creating indexes...")
+            words = self.db[WORDS_COLLECTION]
+            # Create a compound index to ensure uniqueness on word and url
+            words.create_index([("word", 1), ("url", 1)], unique=True)
+            # Create a compound index to easily sort by word and weight
+            words.create_index([("word", 1), ("weight", -1)])
+
+            # Create single field indexes
+            words.create_index("word")
+            words.create_index("url")
+        except Exception as e:
```

Ruff (0.15.9)

[warning] Do not catch blind exception: `Exception`

(BLE001)